source/slang/lookup.cpp

```diff
@@ -224,6 +226,10 @@ void DoLocalLookupImpl(
     LookupResult&		    result,
     BreadcrumbInfo*		    inBreadcrumbs)
 {
+    if (result.lookedupDecls.Contains(containerDeclRef))
```

Comment:
This seems to be another fix for "diamond" inheritance cases, which I guess will arise once we have `extension` decls adding new interface conformances.

We both know that the right long-term solution will be to "linearize" the inheritance DAG for a type/extension to get a list that includes each base type only once...